freetextresponse/freetextresponse.py

```diff
@@ -0,0 +1,486 @@
+from xblock.fields import Scope
+from xblock.fields import String
+from xblock.fragment import Fragment
+from xblock.validation import ValidationMessage
+from xblockutils.studio_editable import StudioEditableXBlockMixin
+
+
+class FreeTextResponse(StudioEditableXBlockMixin, XBlock):
+    #  pylint: disable=too-many-ancestors, too-many-instance-attributes
+    """
+    Enables instructors to create questions with free-text responses.
+    """
+    @staticmethod
+    def workbench_scenarios():
+        """
+        Gather scenarios to be displayed in the workbench
+        """
+        scenarios = [
+            ('FreeText XBlock',
+             '''<sequence_demo>
+                    <freetextresponse />
+                    <freetextresponse name='My First XBlock' />
+                </sequence_demo>
+             '''),
+        ]
+        return scenarios
+
+    display_correctness = Boolean(
+        display_name=_('Display Correctness?'),
+        help=_(
+            'This is a flag that indicates if the indicator '
+            'icon should be displayed after a student enters '
+            'their response'
+        ),
+        default=True,
+        scope=Scope.settings,
+    )
+    display_name = String(
+        display_name=_('Display Name'),
+        help=_(
+            'This is the title for this question type'
+        ),
+        default=_('Free-text Response'),
+        scope=Scope.settings,
+    )
+    fullcredit_keyphrases = List(
+        display_name=_('Full-Credit Key Phrases'),
+        help=_(
+            'This is a list of words or phrases, one of '
+            'which must be present in order for the student\'s answer '
+            'to receive full credit'
+        ),
+        default=[],
+        scope=Scope.settings,
+    )
+    halfcredit_keyphrases = List(
+        display_name=_('Half-Credit Key Phrases'),
+        help=_(
+            'This is a list of words or phrases, one of '
+            'which must be present in order for the student\'s answer '
+            'to receive half credit'
+        ),
+        default=[],
+        scope=Scope.settings,
+    )
+    max_attempts = Integer(
+        display_name=_('Maximum Number of Attempts'),
+        help=_(
+            'This is the maximum number of times a '
+            'student is allowed to attempt the problem'
+        ),
+        default=0,
+        values={'min': 1},
+        scope=Scope.settings,
+    )
+    max_word_count = Integer(
+        display_name=_('Maximum Word Count'),
+        help=_(
+            'This is the maximum number of words allowed for this '
+            'question'
+        ),
+        default=10000,
+        values={'min': 1},
+        scope=Scope.settings,
+    )
+    min_word_count = Integer(
+        display_name=_('Minimum Word Count'),
+        help=_(
+            'This is the minimum number of words required '
+            'for this question'
+        ),
+        default=0,
+        values={'min': 1},
```

Comment:
I agree that it may seem counterintuitive, but this is just how the mixin works. The default value can be anything, as it is not subject to validation. However, once the user picks a value, the value picked must be either the default value or a value that meets the constraints above (min &/or max). Additionally, both min and max must be positive. Hope this clarifies your doubts.